clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3169,7 +3169,9 @@ class SyclKernelIntFooterCreator : public SyclKernelFieldHandler {
     return true;
   }
   bool handleArrayType(FieldDecl *FD, QualType FieldTy) final {
-    Footer.addTypeToCheckForDeviceCopyability(FieldTy);
+    QualType ElementTy = cast<ArrayType>(FieldTy)->getElementType();
```

Comment:
I don't think this handles multi-dimensional arrays correctly.

Reply:
Actually I'm not sure. The visitor might already handle this. Can you check this?

Reply:
Doesn't it depend on whether the SyclKernelintFooterCreator does the 'look into everything' bit?  If the type defines the `VisitInsideSimpleContainers` to be true, the TOP level handleArrayType wouldn't work (since the ElementTy would ALSO be an array type), but when the visitors go into THAT array, it would work?

So you'd get:

`handleArrayType(FieldTy = int[1][2]);`

Then a call to 
`handleArrayType(FieldTy = int[1]);`

?? In further 'retrospect' (current spect? whatever), I think we might not need handleArrayType at all in that case, since the inner-most type should go to handleStructType if necessary.